install_config/topics/configuring_rhv_objects.adoc

```diff
@@ -1,66 +1,410 @@
+bastion_root_password: <bastion_root_password>
+rhsub_user: <Red_Hat_Subscription_Manager_username>
+rhsub_pass: <Red_Hat_Subscription_Manager_password>
+rhsub_pool: <Red_Hat_Subscription_Manager_pool_id>
+root_password: <OpenShift_node_root_password>
+----
+
+. Save the file and record the vault password.
+
+. Create the installation playbook, *_/bastion_installation/create-bastion-machine-playbook.yaml_*, and update its parameters:
++
+[source,yml]
+----
+---
+- name: Create a bastion machine
+  hosts: localhost
+  connection: local
+  gather_facts: false
+
+  vars:
+    engine_url: https://_Manager_FQDN_/ovirt-engine/api
+    engine_user: <admin@internal>
+    engine_password: "{{ engine_password }}"
+    engine_cafile: /etc/pki/ovirt-engine/ca.pem
+
+    qcow_url: <1>
+    template_cluster: Default
+    template_name: rhelguest7
+    template_memory: 4GiB
+    template_cpu: 2
+    wait_for_ip: true
+    debug_vm_create: false
 
-To integrate {product-title} with Red Hat Virtualization, take the following
-actions as part of your
-xref:../install/host_preparation.adoc#install-config-install-host-preparation[host preparation].
+    vms:
+    - name: rhel-bastion
+      cluster: Default
+      profile:
+        cores: 2
+        template: "{{ template_name }}"
+        root_password: "{{ root_password }}"
+        ssh_key: "{{ lookup('file', '/root/.ssh/id_rsa_ssh_ocpadmin.pub') }}"
+        state: running
+      cloud_init:
+        custom_script: |
+          rh_subscription:
+            username: "{{ rhsub_user }}"
+            password: "{{ rhsub_pass }}"
+            auto-attach: true
+            disable-repo: ['*']
+            # 'rhel-7-server-rhv-4.2-manager-rpms' supports RHV 4.2 and 4.3
+            enable-repo: ['rhel-7-server-rpms', 'rhel-7-server-extras-rpms', 'rhel-7-server-ansible-2.7-rpms', 'rhel-7-server-ose-3.11-rpms', 'rhel-7-server-supplementary-rpms', 'rhel-7-server-rhv-4.2-manager-rpms']
+          packages:
+            - ansible
+            - ovirt-ansible-roles
+            - openshift-ansible
+            - python-ovirt-engine-sdk4
+  tasks:
+    - user
+        name: root
 
-. To provide high availability in case of the loss of one hypervisor host,
-add each class of instance to a negative affinity group. See
-link:https://ovirt.org/develop/release-management/features/sla/vm-affinity/[VM Affinity].
+  roles:
+    - oVirt.image-template
+    - oVirt.vm-infra
 
-. To ensure that the {product-title} environment meets the
-xref:../install/prerequisites.adoc#hardware[minimum hardware requirements],
-create templates for virtual machines that use the following resources:
+- name: post installation tasks on the bastion machine
+  hosts: rhel-bastion
+  tasks:
+    - name: create ovirt-engine pki dir
+      file:
+        state: directory
+        dest: /etc/pki/ovirt-engine/
+    - name: Copy the engine ca cert to the bastion machine
+      copy:
+        src: "{{ engine_cafile }}"
+        dest: "{{ engine_cafile }}"
```

Comment:
@rgolangh @apinnick This must be changed. There is no guarantee whatsoever that the variable `engine_cafile` passed by user is the same as our intended target location - `/etc/pki/ovirt-engine`. See previous task. The best solution here would IMHO be to create a new variable at play level (e.g. `ca_destination_dir` or something like this) and then pass such variable to `dest` parameter in this and also previous task.

Reply:
Roy, you decide

Reply:
@rgolangh There's one more problem with this task. The variable `engine_cafile` is defined for the previous play, but not for this one. That's why you'll get undefined variable error here.

Reply:
but engine_cafile should be in secure_vars.yaml and should be used in both invocations.

Reply:
Looks like it got lost. This is what is in secure_vars.yaml:

. Add the following variables to *_secure_vars.yaml_*:
+
[source,yml]
----
engine_password: <Manager_password>
bastion_root_password: <bastion_root_password>
rhsub_user: <Red_Hat_Subscription_Manager_username>
rhsub_pass: <Red_Hat_Subscription_Manager_password>
rhsub_pool: <Red_Hat_Subscription_Manager_pool_id>
root_password: <OpenShift_node_root_password>

Reply:
@rgolangh IMO `engine_cafile` does not belong into secure_vars.yaml. AFAIK we put credentials into the vault. There is no reason to put this there since this is just a simple path to file, nothing too secret. The only reason it is stored in the variable is not to repeat the code. Are you sure that moving it to secure_vars.yaml is the correct way?